case-lib/lib.sh

```diff
@@ -35,7 +35,8 @@ fi
 
 # setup SOFCARD id
 if [ ! "$SOFCARD" ]; then
-    SOFCARD=$(grep -v 'sof-probes' /proc/asound/cards | grep 'sof-[a-z]' | awk '{print $1;}')
+	# remove leading spaces and check if the line starts with a number
+	SOFCARD=$(grep -v 'sof-probes' /proc/asound/cards | awk '/sof-[a-z]/ && $1 ~ /^[0-9]+$/ { $1=$1; print $1;}')
```

Comment:
Nit: you can insert a newline after the `|`, it should be ignored inside `$( )` no need to escape it. In general whenever inserting a newline somewhere in a script passes a simple test then it will pass all tests. There are many traps in shell scripts but not with _unescaped_ newlines: they either work or don't.